config/tectonic-node-controller/config.go

```diff
@@ -0,0 +1,40 @@
+package tnc
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Kind is the TypeMeta.Kind for the TNC config.
+	Kind = "TectonicNodeControllerConfig"
+	// APIVersion is the TypeMeta.APIVersion for the TNC config.
+	APIVersion = "v1"
+)
+
+// TectonicNodeControllerConfig is the config structure TNC to generate NodeConfigs
+type TectonicNodeControllerConfig struct {
+	metav1.TypeMeta       `json:",inline"`
+	HTTPProxy             string `json:"HTTPProxy"`
+	HTTPSProxy            string `json:"HTTPSProxy"`
+	NoProxy               string `json:"NoProxy"`
```

Comment:
These proxy values may be something we want to add later if it really is that painful for a user to add this to their own ignition stub - but for now our goal should be to remove as many parameters as possible.